packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -56,6 +58,7 @@ export interface GraphNGProps extends Themeable2 {
   propsToDiff?: Array<string | PropDiffFn>;
   preparePlotFrame?: (frames: DataFrame[], dimFields: XYFieldMatchers) => DataFrame;
   renderLegend: (config: UPlotConfigBuilder) => React.ReactElement | null;
+  replaceVariables?: InterpolateFunction;
```

Comment:
My only concern is with this prop. I think it shouldn’t be at this low level component but instead in timeseries component. My thinking is that the variable interpolation is a grafana app specific feature that shouldn’t tied to this component. have you tried adding this logic to the timeseries component instead?

Reply:
I thought the same initially, but then I saw that Candlestick was also affected by this issue and thought to make the fix for all existing (and future) panels that use `Timeseries` component. I understand the concern though and agree with it. I'll make the changes and push the modifations.